docs/overview/architecture.md

```diff
@@ -1,10 +1,12 @@
-## LDK Architecture
-![LDK Architecture](../assets/ldk-architecture.svg)
+## Architecture
+![Architecture](../assets/architecture.png)
```

Comment:
Diagram:
* doubt: what is wallet tooling ? 
* instead of direct arrow to database, it should be to Storage Module(dotted) and then to database.
* "LDK creates events that users can respond to" -> "LDK emits events that users can act upon"
* "Net I/O" -> "Network I/O"

Reply:
Not 100% sure what we mean by tooling, I think it can refer to different types of integrations, HWI, remote signing, encryption, testing in different environments etc.

> instead of direct arrow to database, it should be to Storage Module(dotted) and then to database.
We don't really have a Storage Module AFAIK, rather LDK provides the raw bytes for the network graph, channel state etc and the user defines the schema they want to use to back that data up. Correct me if I'm wrong @jkczyz @arik-so .

The rest changes to the diagram seem reasonable, thanks.

Reply:
Yeah I think that's fair, maybe these dotted boxes are better being termed as `components` as you had before. It's tricky some have default implementations others do not. I like the higher-level approach though as it's easier to drill down from there e.g I know to get started I'll need:

 - A `PeerManager`
 - A `ChannelManager`
 - A way to route payments, generate invoices etc

Then I can drill down into what "modules" are needed to create these objects.

Reply:
> So maybe a dotted box is really either a module or set of related modules.

If my memory is correct from when we built this family of diagrams at BTC++, the dotted line was meant to indicate that the developer could swap this out with whatever they wanted. For example, Net I/O could be the pre-built Tokio example, or they could swap that out with their own network I/O stuff. I'm not sure if it makes sense to make that distinction explicit with a key/legend, but I think that was the intent. Many of the other diagrams don't have as many dotted lines.

Here is an updated SVG with revised text and a Persistence box.

![LDK_architecture_revised_2022-08-08](https://user-images.githubusercontent.com/2414177/183464415-996afa34-eb63-4d0f-8ae6-6fc6bca372c0.svg)

Reply:
👍 , Persistence/Storage also needs to be dotted boundary around box ?

Reply:
Yeah, persistence can be swapped out, too.

Reply:
![LDK_architecture_revised_2022-08-09](https://user-images.githubusercontent.com/2414177/183673604-0cfb0f4b-17ba-4695-a94c-8709c263e3c4.svg)

Reply:
LGTM 👍

Reply:
Should we remove the arrow from the dotted line going form LDK to Persistence for consistency with the other dotted lines connected to LDK?

Reply:
Good catch, I think so.

Reply:
@sbddesign one last nit 🚧

Reply:
![LDK_architecture_revised_2022-08-10](https://user-images.githubusercontent.com/2414177/184232831-d25bef5b-0c56-4e2d-91d7-173ff7023efd.svg)